conan/tools/microsoft/toolchain.py

```diff
@@ -187,3 +207,16 @@ def _write_main_toolchain(self, config_filename, condition):
         conan_toolchain = "\n".join(line for line in conan_toolchain.splitlines() if line.strip())
         self._conanfile.output.info("MSBuildToolchain writing {}".format(self.filename))
         save(main_toolchain_path, conan_toolchain)
+
+    def _get_extra_flags(self):
+        # Now, it's time to get all the flags defined by the user
+        cxxflags = self._conanfile.conf.get("tools.build:cxxflags", default=[], check_type=list)
+        cflags = self._conanfile.conf.get("tools.build:cflags", default=[], check_type=list)
+        sharedlinkflags = self._conanfile.conf.get("tools.build:sharedlinkflags", default=[], check_type=list)
+        exelinkflags = self._conanfile.conf.get("tools.build:exelinkflags", default=[], check_type=list)
+        defines = self._conanfile.conf.get("tools.build:defines", default=[], check_type=list)
+        return {
+            "compiler_flags": cflags + cxxflags,
+            "defines": defines,
+            "linker_flags": sharedlinkflags + exelinkflags,
```

Comment:
is it correct to mix shared and exe flags together?
I think the idea is to apply shared flags to .DLL targets only, and exe flags to .EXE targets only.

Reply:
Yeah, I've to figure it out. `MSBuildDeps` was not even using the `exelinkflags`...

Reply:
I cannot see anything about this. They are read as `AdditionalOptions` in `<Link>` section and that's it.